packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1489 @@
+  }) : super(
+    name,
+    value,
+    hidden: hidden,
+    nullDescription: nullDescription,
+    unit: unit,
+    tooltip: tooltip,
+    defaultValue: defaultValue,
+  );
+
+
+  /// Calling [computeValue] is allowed to throw an exception.
+  ///
+  /// If an exception is thrown, the value is set to null and the description
+  /// indicates what exception was thrown.
+  DoubleProperty.lazy(
+    String name,
+    ComputePropertyValueCallback<double> computeValue, {
+    bool hidden: false,
+    String nullDescription,
+    String unit,
+    String tooltip,
+    Object defaultValue: kNoDefaultValue,
+  }) : super.lazy(
+    name,
+    computeValue,
+    hidden: hidden,
+    nullDescription: nullDescription,
+    unit: unit,
+    tooltip: tooltip,
+    defaultValue: defaultValue,
+  );
+
+  @override
+  String numberToString() => object?.toStringAsFixed(1);
+}
+
+/// An int valued property with an optional unit the value is measured in.
+///
+/// Examples of units include 'px' and 'ms'.
+class IntProperty extends _NumProperty<int> {
+  IntProperty(String name, int value, {
+    String nullDescription,
+    bool showName: true,
+    String unit,
+    Object defaultValue: kNoDefaultValue,
+    bool hidden: false,
+  }) : super(
+    name,
+    value,
+    nullDescription: nullDescription,
+    showName: showName,
+    unit: unit,
+    defaultValue: defaultValue,
+    hidden: hidden,
+  );
+
+  @override
+  String numberToString() => object.toString();
+}
+
+/// Property which clamps a [double] to between 0 and 1 and formats it as a
+/// percentage.
+class PercentProperty extends DiagnosticsProperty<double> {
+  PercentProperty(String name, double fraction, {
+    String nullDescription,
+    bool showName: true,
+  }) : super(
+    name,
+    fraction,
+    nullDescription: nullDescription,
+    showName: showName,
+  );
+
+  @override
+  String propertyValueToString() {
+    if (object == null)
+      return object.toString();
+
+    return '${(object.clamp(0.0, 1.0) * 100.0).toStringAsFixed(1)}%';
+  }
+}
+
+/// Property which handles [Matrix4] that represent transforms.
+class TransformProperty extends DiagnosticsProperty<Matrix4> {
+  TransformProperty(String name, Matrix4 value, {
+    Object defaultValue: kNoDefaultValue,
+  }) : super(
+    name,
+    value,
+    defaultValue: defaultValue,
+  );
+
+  @override
+  String propertyValueToString() => debugDescribeTransform(object).join('\n');
+}
+
+/// Property with a bool value described by custom strings.
+///
+/// The property is only shown if there is a flag string matching the bool
+/// value.
+class FlagProperty extends DiagnosticsProperty<bool> {
+  /// Flag only shown when the value is true with a flag name defaulting
+  /// to the property name.
+  ///
+  /// Specify `trueDescription` if the property name is not a good flag
+  /// name. The property name is always hidden as it is redundant with the flag
+  /// name.
+  FlagProperty(
+      String name,
+      bool value, {
```

Comment:
Changed these two boolean arguments to named arguments.
Now some of the `value` arguments are named and some are positional.
WDYT about making all the `value` arguments named to keep the api surface uniform. 
It will make all the property definition code a bit more verbose but it would be a bit more consistent.
Asking as I don't know of a way to do that refactor automatically with Dart so it will take trivially manually modifying all ~1000 call sites.